scripts/apps/authoring/macros/macros.js

```diff
@@ -154,6 +158,12 @@ function MacrosController($scope, macros, desks, autosave, $rootScope, storage)
         $scope.loading = true;
         return macros.call(macro, item).then((res) => {
             if (!res.diff) {
+                Object.keys(res.item || {}).forEach((field) => {
+                    if (isString(res.item[field]) && res.item[field] !== item[field] && field !== 'body_html') {
```

Comment:
For body the html conversion is not working well and I want to prevent it.
For macros that changes body it should return the diff.

Reply:
> For body the html conversion is not working well and I want to prevent it.
How the HTML conversion from `body_html` is different from that of abstract. They are both edito3 fields, aren't they?

> The macros that changes body should return the diff and this will be processed and the body will be updated.
If I understand correctly, the code I'm commenting on works on all editor3 fields, but `body_html` which requires special handling. I'm wondering if that special handling could be applied to all fields in the same way since they all are from editor3. Would that work?

Reply:
>How the HTML conversion from body_html is different from that of abstract. They are both edito3 fields, aren't they?

For other fields(except body) the requirements is just to support simple formatting like bold, italic, underline and link. For this kind of content the html import/export is working well.

>If I understand correctly, the code I'm commenting on works on all editor3 fields, but body_html which requires special handling. I'm wondering if that special handling could be applied to all fields in the same way since they all are from editor3. Would that work?

I think it is not needed. Almost all macros are applied to body, from what I know only one changes other fields. Also the content on other fields is simple and html import/export works well.

Reply:
We shouldn't differentiate between fields. All that works for one editor3 field should work for all since they are using the same data structure. If we already have a function which handles a more complex use-case of `body_html` why not apply that and give all fields the same capability? The code will be simpler as well with one way of handling it.

Reply:
1. It is not needed
2. diff is working for body and we have to change it to work for . all fields -> we have to change all macros and diff related code

Reply:
Why is the estimate so big. Isn't it only about changing field names? Could you like me a function doing macros for `body_html`?

Reply:
https://github.com/superdesk/superdesk-client-core/blob/2db4bfdaa424fddf2f5e2b7b5f64dbd910d85b97/scripts/apps/authoring/macros/macros.js#L260

Reply:
I've just looked at the server code. If I understand correctly, macros can either return diff, processed HTML or do an action(autopublish) and return nothing. It's reasonable then to have 2 ways of processing the response since AAP has a bunch of macros and we may break it if we change the API.

Could you at least move the handling of macros to a single place? Something like:
```
macros.get().then((res) => {
    if (res.diff) {...}
    else if (item.field != res.item.field) {...}
    else {...}
})
```

Reply:
> ... break it if we change the API.

As I already specified I don't want to apply import html on body because it can creates problems.
All macros that changes the body should return diff.

> ... and return nothing

The return nothing case is covered by current implementation.

>Could you at least move the handling of macros to a single place?

Currently the processing that can be done on Macros#call is done and then it triggers the processing in other directives (MacrosReplaceDirective and Editor3Directive).

> else if (item.field != res.item.field) {...}

We have to check that a value is changed for multiple fields not only for one.

Reply:
> As I already specified I don't want to apply import html on body because it can creates problems.

I agree, let's not apply it. What I'm asking is to refactor the handling so all conditions are in one place.

If that doesn't work, at least document the fact that html replacement is not done on `body_html` and that `diff` is only applied to `body_html`, but not other fields. Also add a comment to each function linking the other one.

Reply:
>If that doesn't work, ...
As explained it doesn't work:
>Currently the processing that can be done on Macros#call is done and then it triggers the processing in other directives (MacrosReplaceDirective and Editor3Directive).

I added some documentation.

Reply:
@tomaskikutis Do you have any other questions re: this PR?